RobotFramework_DoIP/DoipKeywords.py

```diff
@@ -0,0 +1,92 @@
+from robot.api.deco import keyword
+from robot.api import logger
+from robot.libraries.BuiltIn import BuiltIn
+
+from doipclient import DoIPClient
+
+class DoipKeywords(object):
+
+    def __init__(self):
+        address, announcement = DoIPClient.get_entity()
+        logical_address = announcement.logical_address
+        ip, port = address
+        print(ip, port, logical_address)
+
+        self.ip = ip
+        self.logical_address = logical_address
+        logger.info(f"IP: {ip} ")
+
+    @keyword
+    def connect(self):
+        pass
+    
+    @keyword
+    def send_diagnostic(self):
+        pass
+    
+    @keyword
+    def receive_diagnostic(self):
+        pass
+
+    @keyword
+    def reconnect(self):
+        pass
+
+    @keyword
+    def disconnect(self):
+        pass
+
+    @keyword("Await Vehicle Annoucement")
+    def await_vehicle_announcement(self):
+        address, announcement = DoIPClient.await_vehicle_announcement()
+        print(address)
+        logical_address = announcement.logical_address
+        ip, port = address
+        print(ip, port, logical_address)
+        return ip, port, logical_address
+
+    @keyword("Get Entity")
+    def get_entity(self):
```

Comment:
Hi Thong,
here I miss parameters e.g. 
ecu_ip_address='255.255.255.255', protocol_version=2, eid=None, vin=None

You need to wrap the whole interface inclusive parameters, that the full functionality is available as robot keywords.

Thank you,
 Thomas

Reply:
Hi Thomas,

I have updated it as methods:
```python
@keyword("Get Entity")
    def get_entity(
        ecu_ip_address="255.255.255.255", 
        protocol_version=0x02, 
        eid=None, 
        vin=None,
    ):
```
If you have any comments, please tell me.

Thank you,
Thong Hua